### PR TITLE
Support ImageFromBytes

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
### Summary

- Addresses: CVS-151819

> [!WARNING]
> There is one major UX issue here. Geti allows passing dataset that do not have pixel-level annotations even though the task is of type segmentation or detection. This means that the task returns pixel-level metrics as 0. This might mislead users.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/data/dataset/anomaly.py [hunk 1]
```diff
@@ -142,6 +131,106 @@ def _get_item_impl(
...
+            id_label_mapping = {}
+            label_items = self.dm_subset.categories()[AnnotationType.label].items
+            for label_item in label_items:
+                if any("normal" in attribute.lower() for attribute in label_item.attributes):
+                    label = AnomalyLabel.NORMAL
+                else:
+                    label = AnomalyLabel.ANOMALOUS
+                id_label_mapping[label_item.name] = label
+            # There is probably a better way to get the label id from datumaro
+            label = id_label_mapping[datumaro_item.attributes["roi"]["labels"][0]["label"]["_id"]]
```
Comment (ashwinvaidya17) [inline src/otx/core/data/dataset/anomaly.py]: I need to access the ids in ROI to get the ids corresponding to the label_item. Just looking at the labels of the arrow dataset does not give me any information 
image
However, the tests fail due to missing 'roi'. Any ideas?

Comment (harimkang) [reply src/otx/core/data/dataset/anomaly.py]: In my opinion, this change should only affect the arrow format - the unit-test doesn't test the arrow format separately. The fact that it fails seems to imply that it also affects non-arrow formats.

Comment (ashwinvaidya17) [reply src/otx/core/data/dataset/anomaly.py]: @wonjuleee

Comment (ashwinvaidya17) [reply src/otx/core/data/dataset/anomaly.py]: @harimkang I am now using a different approach

**Review by harimkang — APPROVED** [review]

**Review by kprokofi — APPROVED** [review]

**Pull request merged**